### DISCUSSION TITLE
Is there a list of all LSP features supported by `pyright-langserver`?

### DISCUSSION THREAD

**Discussion by neelkarma** [microsoft/pyright#1752] [Q&A] [answered]
If there is, can you send a link, and if there isn't, it would be nice to have the list.

**Comment by erictraut** [comment, marked as answer]
I'm not sure what resolution you're looking for. For example, do you want to know whether it supports "hover text" or do you want to know whether that hover text is in the form of plain text or markdown?

It's probably best for you to simply look at the Pyright code. That will provide whatever information you are looking for. This [source file](https://github.com/microsoft/pyright/blob/master/packages/pyright-internal/src/languageServerBase.ts) is where the language server registers for various callbacks such as `onHover`, `onDocumentSymbol`, `onReferences`, `onCompletion`, etc.
